### PR TITLE
Add an add card feature

### PR DESCRIPTION
**Pull request by shawnclq**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddCardCommand.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    public static final String COMMAND_WORD = "addc";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a card to the address book. "
+            + "Parameters: "
+            + PREFIX_FRONT + "FRONT "
+            + PREFIX_BACK + "BACK ";
+
+    public static final String MESSAGE_SUCCESS = "New card added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This card already exists in the address book";
```
Comment (yong-jie) [inline src/main/java/seedu/address/logic/commands/AddCardCommand.java]: Ideally these variables will have to be changed to match the context of our flash card product

File: src/main/java/seedu/address/storage/XmlAdaptedCard.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+//        return new Person(name, phone, email, address, tags);
+//    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof XmlAdaptedPerson)) {
```
Comment (yong-jie) [inline src/main/java/seedu/address/storage/XmlAdaptedCard.java]: might be a bug here. Comparing card object to person object

File: src/main/java/seedu/address/storage/XmlAdaptedCard.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        front = source.getFront();
+        back = source.getBack();
+    }
+
+//    /**
+//     * Converts this jaxb-friendly adapted person object into the model's Person object.
+//     *
+//     * @throws IllegalValueException if there were any data constraints violated in the adapted person
+//     */
+//    public Card toModelType() throws IllegalValueException {
```
Comment (yong-jie) [inline src/main/java/seedu/address/storage/XmlAdaptedCard.java]: this method might need refactoring to match the usage of a flash card

**Comment by jethrokuan** [comment]
Resolves #32 , resolves #14

File: src/test/java/seedu/address/testutil/TypicalCards.java [hunk 4]
```diff
@@ -1,16 +1,68 @@
...
+    public static final Card CARD_2 = new CardBuilder().withFront(VALID_FRONT_CARD_2).withBack(VALID_BACK_CARD_2)
+            .build();
 
     private TypicalCards() {} // prevents instantiation
+
+    /**
+     * Returns an {@code AddressBook} with all the typical persons.
+     */
+    public static AddressBook getTypicalAddressBookWithCards() {
+        AddressBook ab = new AddressBook();
```
Comment (jethrokuan) [inline src/test/java/seedu/address/testutil/TypicalCards.java]: try not to use abbreviations like this. In java it is smelly :P

**Pull request merged**